src/main/java/forestry/plugins/compat/PluginNatura.java

```diff
@@ -120,196 +179,74 @@ public void registerCrates() {
-		}
-		if (logWillow != null) {
-			crateRegistry.registerCrate(new ItemStack(logWillow, 1, 0));
-		}
-
-		Item bloodWood = getItem("bloodwood");
-		if (bloodWood != null) {
-			crateRegistry.registerCrate(new ItemStack(bloodWood, 1, 0));
-		}
-		Item darkTree = getItem("Dark Tree");
-		if (darkTree != null) {
-			crateRegistry.registerCrate(new ItemStack(darkTree, 1, 0));
-			crateRegistry.registerCrate(new ItemStack(darkTree, 1, 1));
-		}
-		Item heatSand = getItem("heatsand");
-		if (heatSand != null) {
-			crateRegistry.registerCrate(new ItemStack(heatSand, 1, 0));
-		}
-		Item taintedSoil = getItem("soil.tainted");
-		if (taintedSoil != null) {
-			crateRegistry.registerCrate(new ItemStack(taintedSoil, 1, 0));
-		}
+		Iterables.concat(
+				edibles, logs, saplings, shrooms,
+				materials, seeds, berries, fruits,
+				crops
+		).forEach(item -> crateRegistry.registerCrate(item));
 	}
 
 	@Override
 	public void registerRecipes() {
 		int amount = ForestryAPI.activeMode.getIntegerSetting("squeezer.liquid.seed");
-
-		Item seed = getItem("barley.seed");
-		if (seed != null) {
-			ItemStack seedBarley = new ItemStack(seed, 1, 0);
-			ItemStack seedCotton = new ItemStack(seed, 1, 1);
-
-			ArrayList<ItemStack> seedList = new ArrayList<>();
-			if (!seedBarley.isEmpty()) {
-				seedList.add(seedBarley);
-				RecipeManagers.moistenerManager.addRecipe(seedBarley, new ItemStack(Blocks.MYCELIUM), 5000);
-			}
-			if (!seedCotton.isEmpty()) {
-				seedList.add(seedCotton);
-			}
-
-			for (ItemStack aSeedList : seedList) {
-				RecipeManagers.squeezerManager.addRecipe(10, aSeedList, Fluids.SEED_OIL.getFluid(amount));
-			}
+		for (ItemStack seed : seeds) {
+			RecipeManagers.squeezerManager.addRecipe(10, NonNullList.from(seed, seed), Fluids.SEED_OIL.getFluid(amount));
+			RecipeManagers.moistenerManager.addRecipe(seed, new ItemStack(Blocks.MYCELIUM), 5000);
 		}
 
 		if (ForestryAPI.enabledPlugins.contains(ForestryPluginUids.FARMING)) {
-			Block cropBlock = getBlock("N Crops");
-			//TODO: Natura for 1.9
-			//			Farmables.farmables.get("farmWheat").add(new FarmableHandPlanted(seedBarley, cropBlock, 3));
-			//			Farmables.farmables.get("farmWheat").add(new FarmableHandPlanted(seedCotton, cropBlock, 8));
-		}
-
-		List<ItemStack> berries = new ArrayList<>();
-		if (!berryBlight.isEmpty()) {
-			berries.add(berryBlight);
-		}
-		if (!berryDusk.isEmpty()) {
-			berries.add(berryDusk);
-		}
-		if (!berrySky.isEmpty()) {
-			berries.add(berrySky);
-		}
-		if (!berrySting.isEmpty()) {
-			berries.add(berrySting);
-		}
-		if (!berryRasp.isEmpty()) {
-			berries.add(berryRasp);
-		}
-		if (!berryBlue.isEmpty()) {
-			berries.add(berryBlue);
-		}
-		if (!berryBlack.isEmpty()) {
-			berries.add(berryBlack);
-		}
-		if (!berryMalo.isEmpty()) {
-			berries.add(berryMalo);
-		}
-
+			cropBlocks.forEach(itemStack -> {
+				Block block = ItemStackUtil.getBlock(itemStack);
+				ItemStack seedItem;
+				int maxAge;
+				try {
+					// XXX: getMaxAge
```

Comment:
XXX is the same as HACK.
Reflection is a hack in my eyes but regarding Minecraft modding in general I guess I can remove that note.

Maybe there is a better way to query custom attributes from items. But as far as I can see that's the easiest one.